sqlitecluster/SQLiteNode.cpp

```diff
@@ -2692,6 +2632,94 @@ bool SQLiteNode::_isQueuedCommandMapEmpty() {
     return true;
 }
 
+void SQLiteNode::_removeQueuedCommand(SQLiteNode::Command* command) {
+    // Iterate in the same order we probably found this command in. I.e., it was most likely found by
+    // getNextQueuedActionableCommand, at the front of a queue from it's perspective, so we'll look in the same order.
+    for (auto it = _queuedCommandMap.rbegin(); it != _queuedCommandMap.rend(); ++it) {
+        
+        auto& list = it->second;
+        auto commandIt = list.begin();
+        while (commandIt != list.end()) {
+
+            // Is this our command?
+            if (command == *commandIt) {
+                // Erase it.
+                list.erase(commandIt);
+
+                // If it was the last command in the whole list, erase that, to.
+                if (list.empty()) {
+                    // Lol. We can't just erase `it` because `erase()` takes an `iterator` and `it` is a
+                    // `reverse_iterator`. Looking up the correct iterator by the priority should be much faster though
+                    // than iterating across potentially many commands in the wrong priority queue to find the one that
+                    // we want to remove.
+                    _queuedCommandMap.erase(_queuedCommandMap.find(it->first));
+                }
+
+                // Ok, we've removed it.
+                return; 
+            }
+
+            ++commandIt;
+        }
+    }
+}
```

Comment:
Can you add an `SERROR()` here?  We should never get here, but it'd be good to know if we did.